handler/core/src/main/java/io/knotx/fragments/handler/action/PayloadToBodyActionFactory.java

```diff
@@ -38,25 +39,32 @@ public String getName() {
   }
 
   @Override
+
   public Action create(String alias, JsonObject config, Vertx vertx, Action doAction) {
-    checkArgument(doAction != null, "Payload to body action does not support doAction");
+    checkArgument(Objects.nonNull(doAction), "Payload to body action does not support doAction");
 
     return (fragmentContext, resultHandler) -> {
       Fragment fragment = fragmentContext.getFragment();
       String payloadKey = Objects.nonNull(config) ? config.getString(KEY) : null;
-
+      ActionLogger actionLogger = ActionLogger.create(alias, config);
       FragmentResult result = getBodyFromPayload(payloadKey, fragment.getPayload())
-          .map(body -> toFragmentResult(fragment, body))
-          .orElse(new FragmentResult(fragment, FragmentResult.ERROR_TRANSITION));
+          .map(body -> toFragmentResult(fragment, body, actionLogger))
+          .orElseGet(() -> toErrorFragmentResult(fragment, payloadKey, actionLogger));
 
       Future<FragmentResult> resultFuture = succeededFuture(result);
       resultFuture.setHandler(resultHandler);
     };
   }
 
-  private FragmentResult toFragmentResult(Fragment fragment, String body) {
+  private FragmentResult toErrorFragmentResult(Fragment fragment, String payloadKey, ActionLogger actionLogger) {
+    actionLogger.error("body", String.format("No value found under key in payload %s", payloadKey));
```

Comment:
Can we agree that there is one field dedicated to error messages?
```
{
  message = "No value..."
  code = Optional
  timestamp = ...
}
```
We had `fallback`, now `body`.